test/error.js

```diff
@@ -292,3 +293,66 @@ test('error.cwd is undefined on failure if not passed as options', async t => {
 	const {cwd} = await t.throwsAsync(execa('fail.js'));
 	t.is(cwd, expectedCwd);
 });
+
+const testUnusualError = async (t, error) => {
+	const childProcess = execa('empty.js');
+	childProcess.emit('error', error);
+	const {message, originalMessage} = await t.throwsAsync(childProcess);
+	t.true(message.includes(String(error)));
+	t.is(originalMessage, String(error));
+};
+
+test('error instance can be null', testUnusualError, null);
+test('error instance can be false', testUnusualError, false);
+test('error instance can be a string', testUnusualError, 'test');
+test('error instance can be a number', testUnusualError, 0);
+test('error instance can be a BigInt', testUnusualError, 0n);
+test('error instance can be a symbol', testUnusualError, Symbol('test'));
+test('error instance can be a function', testUnusualError, () => {});
+test('error instance can be an array', testUnusualError, ['test', 'test']);
+
+test('error instance can be undefined', async t => {
+	const childProcess = execa('empty.js');
+	childProcess.emit('error');
+	await t.throwsAsync(childProcess, {message: 'Command failed: empty.js'});
+});
+
+test('error instance can be a plain object', async t => {
```

Comment:
I have seen this too. For example, `js-yaml` [used to do that](https://github.com/nodeca/js-yaml/commit/acb734f8daf9ce8094b3f48d0b8d92b832a42153) (although those were not plain objects but non-error object instances).